tests/aws/services/lambda_/test_lambda.py

```diff
@@ -276,8 +278,14 @@ def test_assume_role(self, create_lambda_function, aws_client, snapshot, functio
         Unknown whether the 8-character hexadecimal number suffix after "lambda-autogenerated-" follows any pattern.
         """
         lambda_autogen_role = "(?<=lambda-autogenerated-)([0-9a-f]{8})"
+        # avoid any other transformers possible registering reference replacements (especially resource transformer)
+        snapshot.transformers.clear()
         snapshot.add_transformer(
-            snapshot.transform.regex(lambda_autogen_role, "<lambda-autogenerated-role-prefix>")
+            [
+                snapshot.transform.regex(account_id, "1" * 12),
+                snapshot.transform.regex(lambda_autogen_role, "<lambda-autogenerated-role-prefix>"),
+                snapshot.transform.regex(r'(?<=/)[a-zA-Z]{1,2}(?="|@)', "<function-name>"),
```

Comment:
much more elegant than the hacky conditional assertions 🥇 👏👏👏 

We can get rid of the `if function_name_length == 1:` conditional at the end of the test now.